network.go

```diff
@@ -931,7 +931,7 @@ func (n *network) ipamAllocateVersion(ipVer int, ipam ipamapi.Ipam) error {
 	}
 
 	if len(*cfgList) == 0 {
-		if ipVer == 6 {
+		if ipVer == 6 && !n.enableIPv6 {
 			return nil
 		}
```

Comment:
In order to support the `--ipv6-only` scenario, we should bail out here in case of v4:

```
if ipVer == 4 && n.ipv6Only {
    return nil
}
```